src/anndata/_io/specs/lazy_methods.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from pathlib import Path, PurePosixPath
+from types import MappingProxyType
+from typing import TYPE_CHECKING, Any, Literal, overload
+
+import h5py
+import numpy as np
+from scipy import sparse
+
+import anndata as ad
+from anndata.compat import H5Array, H5Group, ZarrArray, ZarrGroup
+
+from .registry import _LAZY_REGISTRY, IOSpec
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+
+
+@overload
+def make_index(
+    *,
+    is_csc: Literal[True],
+    stride: int,
+    shape: tuple[int, int],
+    block_id: tuple[int, int],
+) -> tuple[slice, slice]: ...
+@overload
+def make_index(
+    *,
+    is_csc: Literal[False],
+    stride: int,
+    shape: tuple[int, int],
+    block_id: tuple[int, int],
+) -> tuple[slice]: ...
+
+
+def make_index(
+    *, is_csc: bool, stride: int, shape: tuple[int, int], block_id: tuple[int, int]
+) -> tuple[slice, slice] | tuple[slice]:
+    index1d = slice(
+        block_id[is_csc] * stride,
+        min((block_id[is_csc] * stride) + stride, shape[0]),
+    )
+    if is_csc:
+        return (slice(None, None, None), index1d)
+    return (index1d,)
+
+
+@contextmanager
+def maybe_open_h5(path_or_group: Path | ZarrGroup, elem_name: str):
+    if not isinstance(path_or_group, Path):
+        yield path_or_group
+        return
+    file = h5py.File(path_or_group, "r")
+    try:
+        yield file[elem_name]
+    finally:
+        file.close()
+
+
+_DEFAULT_STRIDE = 1000
+
+
+def compute_chunk_layout_for_axis_shape(
+    chunk_axis_shape: int, full_axis_shape: int
+) -> tuple[int, ...]:
+    n_strides, rest = np.divmod(full_axis_shape, chunk_axis_shape)
+    chunk = (chunk_axis_shape,) * n_strides
+    if rest > 0:
+        chunk += (rest,)
+    return chunk
+
+
+@_LAZY_REGISTRY.register_read(H5Group, IOSpec("csc_matrix", "0.1.0"))
+@_LAZY_REGISTRY.register_read(H5Group, IOSpec("csr_matrix", "0.1.0"))
+@_LAZY_REGISTRY.register_read(ZarrGroup, IOSpec("csc_matrix", "0.1.0"))
+@_LAZY_REGISTRY.register_read(ZarrGroup, IOSpec("csr_matrix", "0.1.0"))
+def read_sparse_as_dask(
+    elem: H5Group | ZarrGroup,
+    _reader,
+    dataset_kwargs: Mapping[str, Any] = MappingProxyType({}),
+):
+    import dask.array as da
+
+    path_or_group = Path(elem.file.filename) if isinstance(elem, H5Group) else elem
+    elem_name = (
+        elem.name if isinstance(elem, H5Group) else PurePosixPath(elem.path).name
```

Comment:
This could be useful wrapped up in a helper. E.g. a function like `elemname(elem) -> str`. It's like `filename` in `file_backing.py`.